runtime/dart_isolate.cc

```diff
@@ -672,6 +668,56 @@ Dart_Isolate DartIsolate::DartIsolateGroupCreateCallback(
       .first;
```

Comment:
When falling back to `--no-enable-isolate-groups` line 665 passes `/*is_root_isolate=*/false`. This can cause a double free (I think):

   * `Isolate.spawn()` creates a new isolate, invoking the code on line 658.
   * That isolate dies at some point
   * The `cleanup_isolate` callback is called. Since `!isolate_data->get()->IsRootIsolate()` we free the `DartIsolate` object.
   * The `cleanup_group` callback is called. We delete the same object again

The reason this happens is that, `is_root_isolate` is different from `is_group_isolate` if `--no-enable-isolate-groups`. (Maybe an extra boolean `is_isolate_group_root` could be used for the memory management)